src/lib.rs

```diff
@@ -495,6 +495,43 @@ impl Client {
         nonce: Option<String>,
         signer: &impl Signer,
         url: &str,
+    ) -> Result<BytesResponse, Error> {
+        let mut retries = 3;
+
+        let mut nonce = nonce;
```

Comment:
I don't think there is a need to rebind `nonce`? We can just define the argument as `mut nonce`?